src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -477,6 +484,51 @@ public AgentSettings LoadSettings()
             }
         }
 
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Info(nameof(ConfigureAutoLogonIfNeeded));
+
+            bool enableAutoLogon = command.GetEnableAutoLogon();
+            if(!enableAutoLogon)
+            {
+                Trace.Info("AutoLogon will not be enabled as per user's input.");
+                return;
+            }
+
+            try
+            {
+                AssertAdminAccess(command);
+                
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                iConfigManager.Configure(command);
+
+                if(iConfigManager.RestartNeeded())
+                {
+                    Trace.Info("AutoLogon is configured for a different user than the current user. Machine needs a restart.");
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartPermission();
+                    if(shallRestart)
+                    {
+                        Trace.Info("Restarting the machine now");
+                        _term.WriteLine("Restarting the machine...");
+                        Process.Start("shutdown.exe", "-r -t 0");
```

Comment:
>0 [](start = 61, length = 1)

do we need to have some time here and not restart immediately. just that user has time to react? also print shutdown -a to abort for user help?